resilience4j-grpc/src/main/java/io/github/resilience4j/grpc/circuitbreaker/client/interceptor/ClientDynamicCircuitBreakerInterceptor.java

```diff
@@ -0,0 +1,44 @@
+/*
+ *  Copyright 2019 Marco Ferrer
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+package io.github.resilience4j.grpc.circuitbreaker.client.interceptor;
+
+import io.github.resilience4j.circuitbreaker.CircuitBreaker;
+import io.github.resilience4j.grpc.circuitbreaker.CircuitBreakerCallOptions;
+import io.github.resilience4j.grpc.circuitbreaker.client.ClientCallCircuitBreaker;
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.ClientCall;
+import io.grpc.ClientInterceptor;
+import io.grpc.MethodDescriptor;
+
+public class ClientDynamicCircuitBreakerInterceptor implements ClientInterceptor {
```

Comment:
I'm not sure if these "Dynamic" circuit breakers should be added in v1.0 of this library. My thought is that they could always be added later if the need arises. Would like to get your thoughts on wether they should be included or not. I did not add them to RateLimiter right now just to simplify the code.

Reply:
What exactly does "dynamic" circuit breaker mean?

Reply:
Means you can "dynamically" add a circuit breaker to an individual call.

Example: maybe you only want to have circuit breakers enabled for certain types of calls.

This test shows how it would be used
https://github.com/resilience4j/resilience4j/pull/1459/files#diff-498b0408a54e91876bfc9c5363b65d44c6f63067c399887bca11d8832466cc11R65

My thinking that this is too complex for this to be needed in this library. I was thinking more along the lines of the 80/20 rule. My "guess" is that most people will not need this complexity and if they do need it, they could decorate it themselves using the interceptors here.